crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -93,10 +95,41 @@ pub enum Policy {
     Volume {
         /// Percentage of the order's volume should be taken as a protocol
         /// fee.
-        factor: f64,
+        factor: Factor,
     },
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Into)]
+pub struct Factor(f64);
+
+impl Factor {
+    /// Convert into a `Factor` capping the value to fit [0, 1)
+    pub fn capped_from(value: f64) -> Self {
+        Self(value.max(0.0).min(0.99999))
```

Comment:
I think it makes sense to not have `capped_from` and only use `try_from()` instead. Usually it's better long term to catch unexpected values early instead of trying to make them work somehow.

Reply:
@MartinquaXD yes, but according to @olgafetisova (see https://github.com/cowprotocol/services/issues/2541#issuecomment-2009211011), we should cap the partner fee.

I agree that capping the partner fee but "errorring" for the other fees is a little bit confusing. What should we do?

Reply:
I would agree with Martin here. Specifically, we don't want to cap all factor values for all variants. We want to cap only partnerFee bps, so IMO this should be done only at the place where we extract partnerFee from appData and nowhere else. 

All other `factor`s should be rejected if outside of allowed range.

Reply:
Also, the caps are different:
1. Generally, for all `factor`s =  1.0
2. For bps is 100bps =  0.01

Reply:
Oh I see, I misunderstood it then. Perfect! :ok_hand: thank you guys!

Reply:
And also, we want the 100bps = 0.01 to be configurable (as autopilot config parameter)

Reply:
> but according to Olga (see https://github.com/cowprotocol/services/issues/2541#issuecomment-2009211011), we should cap the partner fee

Capping the values for partner fee and bps is a different concern, though. I thought the type introduced here is just here to make sure that we have reasonable factors. If we want to cap partner fees and bps there should be separate functions handling the individual capping values that return a `Factor`.

Reply:
> And also, we want the 100bps = 0.01 to be configurable (as autopilot config parameter)

@sunce86 is it ok if I do that as a follow up task?

@MartinquaXD yeah, I agree, I was actually doing the changes as you said, so we think the same :))  the overuse of `from_capped` came from my misunderstanding that the max value for the partner fee was `1.0` instead of `0.01`